### PR TITLE
(PUP-3920) update puppet to hiera 2.0

### PR DESCRIPTION
**Pull request by hlindberg**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ end
 
 gem "puppet", :path => File.dirname(__FILE__), :require => false
 gem "facter", *location_for(ENV['FACTER_LOCATION'] || ['> 2.0', '< 4'])
-gem "hiera", *location_for(ENV['HIERA_LOCATION'] || ['> 1.0', '< 3'])
+gem "hiera", *location_for(ENV['HIERA_LOCATION'] || ['> 2.0', '< 3'])
```
Comment (MikaelSmith) [inline Gemfile]: `> 2.0`?

Comment (hlindberg) [inline Gemfile]: doh

File: ext/project_data.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ gem_default_executables: 'puppet'
 gem_forge_project: 'puppet'
 gem_runtime_dependencies:
   facter: ['> 2.0', '< 4']
-  hiera: ['> 1.0', '< 3']
+  hiera: ['> 2.0', '< 3']
```
Comment (hlindberg) [inline ext/project_data.yaml]: doh here as well

**Comment by kylog** [comment]
Target stable please.

**Pull request closed without merging**